### ISSUE TITLE
Watch history is not saved after stopping NewPipe

### ISSUE THREAD

**Issue by Sidd065** [TeamNewPipe/NewPipe#11084]
### Checklist

- [x] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.27.0

### Steps to reproduce the bug

1. Click on and start watching a video.
2. Swipe up and stop NewPipe from Recent apps menu.
3. Open NewPipe and watch the same video again

### Expected behavior

The video's watch history will be saved and it will resume from when NewPipe was stopped.

### Actual behavior

The video's watch history isn't saved and the video resumes from where it started at step 1. If the video is paused before stopping NewPipe, watch history get updated.

### Screenshots/Screen recordings

https://github.com/TeamNewPipe/NewPipe/assets/78248734/faad1123-82b6-4090-9f7b-6402d0c1fbd5

### Affected Android/Custom ROM version

Android 13

### Affected device model

Asus ROG Phone 5s

### Additional information

This is a device specific issue that happens on my ROG Phone 5s but not on my other Android 12 phone. The issue still persists after settings NewPipe's battery usage to Unrestricted.

**Comment by Sidd065** [comment]
> Both have exact same Newpipe settings?
Yes both were tested with a fresh install of NewPipe with default settings.

**Comment by Sidd065** [comment]
U

**Comment by ShareASmile** [comment]
We would like to club together the resume not working in intended way in the app, finding the cause of not saving the state would be easy for the developers that way.

**Comment by youreTheCommitToMyPR** [comment]
Same here at #12480 GrapheneOS, Android 16, exploit protection enabled. Maybe something to do with cache? It always happens when I return to the app after a while either from the same or another user profile.
